modules/swagger-promote-core/src/main/java/com/axway/apim/apiimport/actions/CreateNewAPI.java

```diff
@@ -111,7 +111,8 @@ public void execute(APIChangeState changes, boolean reCreation) throws AppExcept
 			new ManageClientApps(changes.getDesiredAPI(), createdAPI, changes.getActualAPI()).execute(reCreation);
 			
 			// V-Host must be managed almost at the end, as the status must be set already to "published"
-			vHostHandler.handleVHost(changes.getDesiredAPI(), createdAPI);
+			//vHostHandler.handleVHost(changes.getDesiredAPI(), createdAPI);
+			// Vhost should be setup during the api publish
```

Comment:
@rathnapandi, I think this is required on versions before 7.6.2 SP3 where is was not possible to set the V-Host on an unpublished API.

Reply:
@cwiechmann vhost should be setup during publish API call. check out the API spec

http://apidocs.axway.com/swagger-ui/index.html?productname=apimanager&productversion=7.6.2&filename=api-manager-V_1_3-swagger.json#!/API_Proxy_Registration/post_proxies_id_publish

Reply:
Removed dead code and just left a comment. 
I also adjusted the test-case to reproduce the issue you experienced. With the change you proposed the error doesn't happen anymore and all tests, incl. the improved test, are running fine.
I assume this code became obsolete as the V-Host became a criteria. With that, the V-Host cannot be updated anymore like any other property and therefore the handling is a different one.